routeros/resource_ipv6_neighbor_discovery.go

```diff
@@ -0,0 +1,153 @@
+		},
+		"mtu": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Description:  "The flag indicates whether hosts should use stateful autoconfiguration (DHCPv6) to obtain addresses",
+			ValidateFunc: validation.IntBetween(0, 4294967295),
+		},
+		"other_configuration": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "The flag indicates whether hosts should use stateful autoconfiguration to obtain additional information (excluding addresses).",
+		},
+		"pref64_prefixes": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Specify IPv6 prefix or list of prefixes within /32, /40. /48, /56, /64, or /96 subnet that will be provided to hosts as NAT64 prefixes.",
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsCIDR,
+			},
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"ra_delay": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The minimum time allowed between sending multicast router advertisements from the interface.",
+			Default:     "3s",
+		},
+		"ra_interval": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The min-max interval allowed between sending unsolicited multicast router advertisements from the interface.",
+			Default:     "3m20s-10m",
+		},
+		"ra_preference": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Specify the router preference that is communicated to IPv6 hosts through router advertisements." +
+				"The preference value in the router advertisements enables IPv6 hosts to select a default router to reach a remote destination",
+			Default:      "medium",
+			ValidateFunc: validation.StringInSlice([]string{"low", "medium", "high"}, false),
+		},
+		"ra_lifetime": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Specify the router preference that is communicated to IPv6 hosts through router advertisements." +
+				"The preference value in the router advertisements enables IPv6 hosts to select a default router to reach a remote destination",
+			Default: "30m",
+		},
+		"reachable_time": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Specify the router preference that is communicated to IPv6 hosts through router advertisements." +
+				"The preference value in the router advertisements enables IPv6 hosts to select a default router to reach a remote destination",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"retransmit_interval": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "The time between retransmitted Neighbor Solicitation messages." +
+				"Used by address resolution and the Neighbor Unreachability Detection algorithm (see Sections 7.2 and 7.3 of RFC 2461)",
+			DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```

Comment:
I found the problem. It is contained in ```DiffSuppressFunc```.
JSON generation in the current implementation:
```
'retransmit-interval':''
'interface':'ether1'
'ra-delay':'3s'                                                                                                                                                                                                                                                                                                               
'ra-interval':'3m20s-10m'                                                                                                                                                                                                                                                                                                     'ra-lifetime':'30m'                                                                                                                                                                                                                                                                                                           
'reachable-time':'10m'                                                                                                                                                                                                                                                                                                        'ra-preference':'high'                                                                                                                                                                                                                                                                                                        
```
JSON with our nice and working ```AlwaysPresentNotUserProvided``` function
```
'retransmit-interval':'12m'
'interface':'ether1'
'ra-delay':'3s'
'ra-interval':'3m20s-10m'
'reachable-time':'10m'
'ra-preference':'high'
'ra-lifetime':'30m'
```
We should remove all such functions if they give such problems. If I remember correctly, we used them somewhere....

Reply:
Thank you very much... I left this here by accident while I was testing something. My apologies, I'll fix it soon.